networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1220 @@
+        if float, is the coefficient of exp(Ax) sampled between (0, 1) and rescaled appropriately.
+        if a lambda expression, determines a custom distribution sampled between (0, 1) and rescaled appropriately.
+
+    Return
+    -------
+    tuple of floats:
+        (treeness, feedforwardness, orderability) hierarchy coordinates
+
+    Examples
+    ---------
+    >>> import numpy as np
+    >>> a = np.array([
+    ...     [0, 0.2, 0, 0, 0],
+    ...     [0, 0, 0, 0.7, 0],
+    ...     [0, 0.4, 0, 0, 0],
+    ...     [0, 0.3, 0.1, 0, 1.0],
+    ...     [0, 0, 0, 0, 0],
+    ... ])
+    >>> b = np.array([
+    ...     [0, 0, 1, 0, 0, 0, 1],
+    ...     [0, 0, 1, 0, 0, 0, 0],
+    ...     [0, 0, 0, 0, 1, 0, 0],
+    ...     [0, 0, 1, 0, 1, 0, 0],
+    ...     [0, 0, 0, 0, 0, 1, 0],
+    ...     [0, 0, 0, 1, 0, 0, 1],
+    ...     [0, 0, 0, 0, 0, 0, 0],
+    ... ])
+    >>> print('(a) Treeness: {0} | Feedforwardness: {1} | Orderability: {2}'.format(*np.round(nx.hierarchy_coordinates(a), 2)))
+    (a) Treeness: -0.04 | Feedforwardness: 0.87 | Orderability: 0.81
+    >>> print('(b) Treeness: {0} | Feedforwardness: {1} | Orderability: {2}'.format(*np.round(nx.hierarchy_coordinates(b), 2)))
+    (b) Treeness: -0.5 | Feedforwardness: 0.56 | Orderability: 0.43
+
+    Notes
+    ______
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+    returns the averaged hierarchy coordinates given the adjacency matrix A
+
+    See Also
+    --------
+    feedforwardness, orderability, treeness
+    """
+    import numpy as np
+
+    np_A = []
+    if (
+        isinstance(A, nx.DiGraph)
+        or isinstance(A, nx.Graph)
+        or isinstance(A, nx.MultiGraph)
+        or isinstance(A, nx.MultiDiGraph)
+    ):
+        np_A = nx.to_numpy_array(A)
+    elif isinstance(A, np.ndarray):
+        np_A = A
+    else:
+        print(
+            "A must be given as either a networkx graph or adjacency matrix (as nested list or 2d numpy array)"
+        )
+
+    if np.array_equal(np.unique(np_A), [0]):  # null check
+        # raise Exception("Unconnected graph; trivially 0 hierarchy")  # TODO: Raise exception if undefined instead
+        return 0, 0, 0
+
+    o, f, t = 0, 0, 0
+    condensed_graphs, original_graphs = node_weighted_condense(
+        A=np_A,
+        num_thresholds=num_thresholds,
+        threshold_distribution=threshold_distribution,
+    )
+    for index in range(len(condensed_graphs)):
+        t += treeness(condensed_graphs[index])
+        f += feedforwardness(condensed_graphs[index])
+        o += orderability(original_graphs[index], condensed_graphs[index])
+    t /= len(condensed_graphs)
+    f /= len(condensed_graphs)
+    o /= len(condensed_graphs)
+    return t, f, o
```

Comment:
I left out the list comprehension here so that I could feed orderability the original graphs directly, as I think that's often faster than re-deriving the original (unweighted) graphs, seen in:
`o += orderability(original_graphs[index], condensed_graphs[index])`

Reply:
I left out the list comprehension here so that I could feed orderability the original graphs directly, as I think that's often faster than re-deriving the original (unweighted) graphs, seen in:
`o += orderability(original_graphs[index], condensed_graphs[index])`

Reply:
Thanks -- I missed that...   My suggestion should read:

```python
o = sum(orderability(OG, CG) for OG, CG in zip(original_graphs, condensed_graphs)) / len(condensed_graph)
```